meta/fraction-metadata/src/main/java/org/wildfly/swarm/fractions/scanner/FilePresenceScanner.java

```diff
@@ -22,12 +22,13 @@
 import java.util.function.Consumer;
 
 import org.wildfly.swarm.spi.meta.FileDetector;
+import org.wildfly.swarm.spi.meta.FileSource;
 import org.wildfly.swarm.spi.meta.FractionDetector;
 
 /**
  * @author Heiko Braun
  */
-public class FilePresenceScanner implements Scanner<String> {
+public class FilePresenceScanner implements Scanner<FileSource> {
     @Override
     public String extension() {
         return XML;
```

Comment:
Hmmm I don't quite understand your question. You mean extension() shouldn't be there? FileSource?